sky/global_user_state.py

```diff
@@ -274,13 +311,35 @@ def get_cluster_from_name(
         return record
 
 
+def update_cost_of_clusters():
+    rows = _DB.cursor.execute(
+        'select * from clusters order by launched_at desc')
+    status_time = int(time.time())
+    for (name, launched_at, handle, _, status, _, metadata, _) in rows:
+        handle = pickle.loads(handle)
+        metadata = json.loads(metadata)
+
+        if handle and metadata:
+            cost_before_start = metadata['cost_before_start']
+            cost_from_start_to_now = 0
+            if ClusterStatus[status] == ClusterStatus.UP:
+                cost_from_start_to_now = get_cost_for_time_interval(
+                    handle, launched_at, status_time)
+            latest_cost = cost_before_start + cost_from_start_to_now
+            metadata['latest_queried_cost'] = latest_cost
+            set_cluster_metadata(name, metadata)
+
+
 def get_clusters() -> List[Dict[str, Any]]:
+    update_cost_of_clusters()
```

Comment:
Related to above comment - perhaps we should not be updating costs in the db and instead computing it on the fly from `usage_intervals` after fetching it from the db.